c/datasetHeartbeatService.c

```diff
@@ -0,0 +1,110 @@
+
```

Comment:
Do you really need a separate source file and installer for this service? Have a look at `c/dataService.c`. Perhaps you can extend that.